dev-python/influxdb: add Python 3.7

Hawk777 opened the pull request.

Hawk777 commented:
A comment on <https://github.com/influxdata/influxdb-python/issues/794> claims that this package has been tested on Python 3.7. I also did some basic things myself which worked. I couldn’t get the full test suite passing, but have filed <https://github.com/influxdata/influxdb-python/issues/799> and <https://github.com/influxdata/influxdb-python/issues/798> related to issues I ran into trying to do so.

mgorny commented on dev-python/influxdb/influxdb-5.2.3.ebuild:
```diff
@@ -0,0 +1,39 @@
+# Copyright 1999-2020 Gentoo Authors
```
Please look up on how to write good commit messages.

Hawk777 replied:
I was under the impression that version bumps in the Gentoo repo typically didn’t say anything about what changed in the underlying package, and that you had to go upstream to find that out. But sure, I’ve added a little bit more text in beea73e70534; let me know if there is anything further you feel is worth putting in the message.

mgorny replied:
But they usually do say what the new version is.

mgorny commented on dev-python/influxdb/influxdb-5.2.3.ebuild:
```diff
@@ -0,0 +1,39 @@
+		dev-python/requests-mock[${PYTHON_USEDEP}]
+	) )"
+
+python_test() {
```
Use `distutils_enable_tests`.

Hawk777 replied:
beea73e70534

mgorny commented on dev-python/influxdb/influxdb-5.2.3.ebuild:
```diff
@@ -0,0 +1,39 @@
+	>=dev-python/requests-2.17.0[${PYTHON_USEDEP}]
+"
+DEPEND="${RDEPEND}
+	dev-python/setuptools[${PYTHON_USEDEP}]
```
Remove the redundant dep.

Hawk777 replied:
beea73e70534

Hawk777 commented:
Nope, and I [reported it upstream](https://github.com/influxdata/influxdb-python/issues/798).

Hawk777 commented:
Hi, just checking if anything further is needed? AFAICT I responded to all questions/comments. Thanks!

juippis commented:
Well this package is gone after 7622f900c65

It needs to be re-added preferably after upstream has fixed reported issues, or put to ::guru for now.
(I guess?)

Hawk777 commented:
So you object to the addition of a version bump with failing tests? I have to wait until another release, hopefully with the tests fixed?

juippis commented:
Truthfully I was just doing some cleaning in Github and noticed this package has been removed, checked in, saw some errors and link to upstream. I have no idea why python team hasn't merged this and didn't dig deeper into it either. 

In **my** eyes adding it now counts as a "new package" and new package additions should be flawless. But python team can decide otherwise.

Hawk777 commented:
> should be flawless

It seems that “package has failing tests” came up on `gentoo-dev` recently, too. I can always just change the ebuild to `RESTRICT=test` if you want? Or add a pile of patches, I guess, though I do prefer sticking close to upstream.

Hawk777 commented:
I’ve rebased this. Still awaiting comment on if any further work is needed, or what that work should be.

Hawk777 commented:
In the rebase I also changed myself to maintainer, since apparently this PR has transformed itself from a version bump into a new package.

Hawk777 commented:
@mgorny @juippis I rebased this and changed it to version 5.3.0 (which upstream released since I started this PR initially). That upstream version change added `dev-python/msgpack` as a dependency (which I added to the ebuild, of course) and fixed the test suite problem I indicated above. However the test suite is now broken for [a different reason](https://github.com/influxdata/influxdb-python/issues/714), but that was easy to work around. The test suite passes for me.

I have not added 3.8. I really tried to, but in order to add 3.8 to this package I would have to add 3.8 to `dev-python/pandas` (because the test suite uses it), in order to add 3.8 to `dev-python/pandas` I would have to add 3.8 to `dev-python/openpyxl`, and GH-14481 already did that in January but has not been merged yet, and it in turn depends on GH-14349 which was also filed in January and not merged yet. I didn’t explore the rest of the deptree of Pandas to see if there were any other packages that would also need bumping.
